File: BasicRxJavaSample/app/src/main/java/com/example/android/observability/demoTest/ImitateInjection.java
package com.example.android.observability.demoTest;

import android.content.Context;

import com.example.android.observability.demoTest.persisitence.ImitateUsersDatabase;
import com.example.android.observability.demoTest.persisitence.ImitateLocalUserDataSource;
import com.example.android.observability.demoTest.ui.ImitateViewModelFactory;

public class ImitateInjection {

    public static ImitateViewModelFactory provideImitateViewModelFactory(Context context) {
        return new ImitateViewModelFactory(provideLocalUserDataSource(context));
    }

    public static ImitateLocalUserDataSource provideLocalUserDataSource(Context context) {
        final ImitateUsersDatabase usersDatabase = ImitateUsersDatabase.getInstance(context);
        return new ImitateLocalUserDataSource(usersDatabase.imitateUserDao());
    }
}

File: BasicRxJavaSampleKotlin/app/src/main/java/com/example/android/observability/imitate/ImitateInjection.kt
package com.example.android.observability.imitate

import android.content.Context
import com.example.android.observability.imitate.persistance.ImitateLocalDataSource
import com.example.android.observability.imitate.persistance.ImitateUserDataBase
import com.example.android.observability.imitate.ui.ImitateViewModelFactory

class ImitateInjection(){

    companion object{
        public fun providerFactory(context: Context):ImitateViewModelFactory{
            return ImitateViewModelFactory(providerUserData(context))
        }
        private fun providerUserData(context: Context):ImitateLocalDataSource{
            return ImitateLocalDataSource(ImitateUserDataBase.getInstance(context))
        }
    }
}